lib/matplotlib/pyplot.py

```diff
@@ -1228,11 +1249,29 @@ def subplot(*args, **kwargs):
         (ax for ax in fig.axes
          if hasattr(ax, 'get_subplotspec') and ax.get_subplotspec() == key),
         None)
+    # if we found an axes at the right position sort out if we
+    # should re-use it
+    if ax is not None:
+        # sort out what projection and class would be created so we
+        # can tell if we should re-use an existing one
+        p_class, pkw = fig._process_projection_requirements(*args, **kwargs)
+        # if no projection or kwargs were passed, return the axes we
+        # found.
+        if projection is proj_placeholder and pkw == {}:
+            ax = ax
+        # if the axes class and kwargs are identical reuse the Axes
+        elif getattr(ax, '_projection_init', ()) == (p_class, pkw):
+            ax = ax
+        # otherwise we need to make a new one!
+        else:
+            ax = None
 
-    # If no existing axes match, then create a new one.
+    # if we do not want to re-use the existing Axes,
     if ax is None:
         ax = fig.add_subplot(*args, **kwargs)
 
+    fig.sca(ax)
```

Comment:
was the call to sca just missing previously in the reuse axes case?

Reply:
yes

Reply:
More verbosely, this behavior used to be implemented in the Figure methods but now needs to be done here instead.